src/vector_transport.jl

```diff
@@ -587,165 +654,326 @@ end
+    vector_transport_to(M::AbstractManifold, p, X, q, m::AbstractVectorTransportMethod)
+    vector_transport_to(M::AbstractManifold, p, X, q, m::AbstractVectorTransportMethod)
 
 Transport a vector `X` from the tangent space at a point `p` on the [`AbstractManifold`](@ref) `M`
-along the [`shortest_geodesic`](@ref) to the tangent space at another point `q`.
-By default, the [`AbstractVectorTransportMethod`](@ref) `method` is
-[`default_vector_transport_method`](@ref)`(M)`.
+along a curve implicitly given by an [`AbstractRetractionMethod`](@ref) associated to `m`.
+By default `m` is the [`default_vector_transport_method`](@ref)`(M)`.
+To explicitly specify a (different) retraction to the implicitly assumeed retraction, see [`VectorTransportTo`](@ref).
+Note that some vector transport methods might also carry their own retraction they are associated to,
+like the  [`DifferentiatedRetractionVectorTransport`](@ref) and some are even independent of the retraction, for example the [`ProjectionTransport`](@ref).
+
+This method is equivalent to using ``d = \operatorname{retr}^{-1}_p(q)`` in [`vector_transport_direction`](@ref)`(M, p, X, q, m, r)`,
+where you can find the formal definition. This is the fallback for [`VectorTransportTo`](@ref).
 """
-function vector_transport_to(M::AbstractManifold, p, X, q)
-    return vector_transport_to(M, p, X, q, default_vector_transport_method(M))
-end
 function vector_transport_to(
     M::AbstractManifold,
     p,
     X,
     q,
-    method::AbstractVectorTransportMethod,
+    m::AbstractVectorTransportMethod = default_vector_transport_method(M),
 )
-    Y = allocate_result(M, vector_transport_to, X, p, q)
-    vector_transport_to!(M, Y, p, X, q, method)
-    return Y
+    return _vector_transport_to(M, p, X, q, m)
+end
+function _vector_transport_to(M::AbstractManifold, p, X, q, m::VectorTransportTo)
+    d = inverse_retract(M, p, q, m.inverse_retraction)
+    return vector_transport_direction(M, p, X, d, m.vector_transport)
+end
+function _vector_transport_to(M::AbstractManifold, p, X, q, ::ParallelTransport)
+    return parallel_transport_to(M, p, X, q)
+end
+function _vector_transport_to(
+    M::AbstractManifold,
+    p,
+    X,
+    q,
+    m::DifferentiatedRetractionVectorTransport,
+)
+    return vector_transport_to_diff(M, p, X, q, m.retraction)
+end
+function vector_transport_to_diff(M::AbstractManifold, p, X, q, r)
+    Y = allocate_result(M, vector_transport_to, X, p)
+    return vector_transport_to_diff!(M, Y, p, X, q, r)
+end
+function _vector_transport_to(M::AbstractManifold, p, X, q, ::ProjectionTransport)
+    return vector_transport_to_project(M, p, X, q)
+end
+function vector_transport_to_project(M::AbstractManifold, p, X, q)
+    Y = allocate_result(M, vector_transport_to, X, p)
+    return vector_transport_to_project!(M, Y, p, X, q)
+end
+function vector_transport_to_project!(M::AbstractManifold, Y, p, X, q)
+    return project!(M, Y, q, X)
```

Comment:
Should this maybe call `embed` on `X` like the projection VT on the fixed rank manifold?

Reply:
No, not in general I think (though I am just playing around unifying embed/checksize and vector checks) – since X is assumed to have a format that first project.
For Fixedrank this is not the case (in SVDMPoint Form) that is why it embeds (basically forming X in the embedding) before projection. So only for Fixed Rank where we have a special (local/per tangent space) representation, embed makes sense.

Reply:
Wouldn't
```julia
function vector_transport_to_project!(M::AbstractManifold, Y, p, X, q)
    return project!(M, Y, q, embed(M, p, X))
end
```
be correct in all cases though? Don't we assume that `embed` is the identity for non-embedded manifolds or manifolds with the same representation as the embedding? I'm mostly thinking about not having to repeat this method for each manifold where embedding is necessary, or correcting my mental model of embedding if it is.

Reply:
we could introduce it as a kind of default? I do not have many in mind where this is different, so if documented properly, that would work. I just feel its not 100% correct, but ok.

Reply:
I'm also not 100% this is how it's supposed to work so maybe let's not touch it here.